### PR TITLE
feat: ogImage component for blog

### PR DESCRIPTION
**Pull request by preschian**
- deprecate `prerender-opengraph` and `prerender` workers
- move `/blog/:slug` on `opengraph` workers to `ssr-opengraph`
- add opengraph component for blog on `ogi` workers

http://localhost:3000/blog/understanding-polkadot-a-deep-dive-into-the-innovative-multi-chain-protocol/__og_image__/og.png

image

### CODE REVIEW

File: ssr-opengraph/prettier.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/** @type {import("prettier").Config} */
```
Comment (roiLeo) [inline ssr-opengraph/prettier.config.js:1]: If possible `.prettierrc` with `semi: false`? in a follow up?

File: prerender/wrangler.toml [hunk 2]
```diff
@@ -3,17 +3,8 @@ main = "src/worker.ts"
 compatibility_date = "2023-05-18"
```
Comment (roiLeo) [inline prerender/wrangler.toml:3]: remove this file if not used?

File: ogi/components/OgImage/blog.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+});
+
+defineProps<{
+  title: string;
+  image: string;
+}>();
+
+const cover: CSSProperties = {
+  objectFit: 'cover',
+  objectPosition: 'center',
```
Comment (roiLeo) [inline ogi/components/OgImage/blog.vue:16]: have you tried `object-cover object-center`?

File: ogi/utils/handler.ts [hunk 4]
```diff
@@ -57,3 +57,25 @@ export const getNftById = async (prefix: Prefix, id: string) => {
...
+
+const encodeEndpoint = (endpoint: string) => {
+  return endpoint.replace(/[:,._/]/g, '-');
+};
+
+export const parseImage = async (imagePath: string) => {
+  const rawImage = `https://raw.githubusercontent.com/kodadot/nft-gallery/main/public${imagePath}`;
+  const encodeImage = encodeEndpoint(rawImage);
+
+  await fetch(`https://image-beta.w.kodadot.xyz/type/url?endpoint=${rawImage}`);
```
Comment (roiLeo) [inline ogi/utils/handler.ts:78]: is it used?

**Review by roiLeo — APPROVED** [review]

File: ogi/components/OgImage/blog.vue (hunk 3 above)
Comment (preschian) [reply ogi/components/OgImage/blog.vue:16]: let met check, hopefully Satori support that

File: ssr-opengraph/prettier.config.js (hunk 1 above)
Comment (preschian) [reply ssr-opengraph/prettier.config.js:1]: updated

File: prerender/wrangler.toml (hunk 2 above)
Comment (preschian) [reply prerender/wrangler.toml]: updated

File: ogi/components/OgImage/blog.vue (hunk 3 above)
Comment (preschian) [reply ogi/components/OgImage/blog.vue:16]: satori still not support that class

**Review by roiLeo — APPROVED** [review]

**Comment by vikiival** [comment]
Please for the fifth time: always do separate PR for the removals :) 

I will approve this one, but next time please to that separately

**Review by vikiival — APPROVED** [review]

**Comment by preschian** [comment]
> Please for the fifth time: always do separate PR for the removals :)

noted, feel free to request change so I can make a revert commit for that

**Comment by preschian** [comment]
deployed to canary

image

**Comment by preschian** [comment]
deployed to production

image

**Pull request merged**